packages/next-app/src/app/stableswapsimulator/(components)/StableCurve.tsx

```diff
@@ -1,32 +1,164 @@
+
+  if (
+    !baselineData ||
+    !baselineData.swapFee ||
+    !baselineData.ampFactor ||
+    !baselineData.tokens
+  )
+    return <Spinner />;
+
+  const [formattedPoints, setFormattedPoints] = useState<{
+    x: string;
+    y: string;
+  }>();
+
+  const tokensSymbol = baselineData.tokens.map((token) => token.symbol);
+
+  const initialAmountsAnalysisTokenIn = calculateAmounts({
+    balance: baselineData?.tokens?.[indexAnalysisToken]?.balance,
+  });
+
+  const initialAmountsAnalysisTokenOut = initialAmountsAnalysisTokenIn.map(
+    (amount) => -1 * amount //This is being multiplied by -1 because it is the OUT part of the trade
   );
-  const amountsTabToken = calculateAmounts(
-    variantData?.tokens?.[indexAnalysisToken]?.balance
+
+  const initialPoolPairDataAnalysisIn = preparePoolPairData({
+    indexIn: indexAnalysisToken,
+    indexOut: indexCurrentTabToken,
+    swapFee: baselineData?.swapFee,
+    allBalances: baselineData?.tokens?.map((token) => token.balance),
+    amp: baselineData?.ampFactor,
+  });
+
+  const initialAmountTabTokenOut = initialAmountsAnalysisTokenIn.map(
+    (amount) => {
+      return (
+        StableMath._exactTokenInForTokenOut(
+          numberToOldBigNumber(amount),
+          initialPoolPairDataAnalysisIn
+        ).toNumber() * -1 //This is being multiplied by -1 because it is the OUT part of the trade
+      );
+    }
   );
+
+  const initialPoolPairDataAnalysisOut = preparePoolPairData({
+    indexIn: indexCurrentTabToken,
+    indexOut: indexAnalysisToken,
+    swapFee: baselineData?.swapFee,
+    allBalances: baselineData?.tokens?.map((token) => token.balance),
+    amp: baselineData?.ampFactor,
+  });
+
+  const initialAmountTabTokenIn = initialAmountsAnalysisTokenIn.map(
+    (amount) => {
+      return StableMath._exactTokenInForTokenOut(
+        numberToOldBigNumber(amount),
+        initialPoolPairDataAnalysisOut
+      ).toNumber();
+    }
+  );
+
   return (
-    <>
-      {amountsAnalysisToken} {amountsTabToken} {indexCurrentTabToken}
-    </>
+    <div className="text-white">
+      <Plot
+        data={[
+          {
+            x: initialAmountsAnalysisTokenIn,
+            y: initialAmountTabTokenOut,
+            type: "scatter",
+            mode: "lines",
+            marker: { color: blueDarkA.blueA9 },
+            hovertemplate: `Swap $%{x:.2f} ${tokensSymbol[indexAnalysisToken]} for ${formattedPoints?.y} ${tokensSymbol[indexCurrentTabToken]} <extra></extra>`,
+            showlegend: false,
+          },
+          {
+            x: initialAmountsAnalysisTokenOut,
+            y: initialAmountTabTokenIn,
+            type: "scatter",
+            mode: "lines",
+            marker: { color: blueDarkA.blueA9 },
+            hovertemplate: `Swap $%{y:.2f} ${tokensSymbol[indexCurrentTabToken]} for ${formattedPoints?.x} ${tokensSymbol[indexAnalysisToken]}<extra></extra>`,
+            showlegend: false,
+          },
+        ]}
```

Comment:
would add both markers in the same trace and show the legend, because the user can click on it to hide it for example.

Reply:
cool, can we update the figma then? or specify this on linear? Because I just followed the proposed directions from both